### PR TITLE
make _readonly feature work with gardener

### LINKED ISSUE #1191: Make _readonly feature work with gardener
**What would you like to be added**:
As of now, we can not use the `_readonly` feature together with `gardener` because `_readonly` provides dm-verity which also uses OverlayFS for `/etc` and `/var`. However, containerd and Docker do not work with OverlayFS. Therefore, we must provide a dedicated `/var` partition which is read writable and is able to be resized.

**Why is this needed**:
We will need dm-verity with `gardener` in future, so we must make sure that both feature work together.

### PR DESCRIPTION
**Pull request by chrinorse**
**How to categorize this PR?**

/kind enhancement
/area os
/os garden-linux

**What this PR does / why we need it**:
make _readonly feature work with gardener
* add empty resizable /var partition
* add systemd-tmpfiles configuration for files under /var

**Which issue(s) this PR fixes**:
Fixes #1191 and #1285

**Special notes for your reviewer**:

**Release note**:

```feature user

```

### CODE REVIEW

**Comment by chrinorse** [comment]
This is on hold, since it is not planned for the next release. Also https://github.com/gardenlinux/gardenlinux/issues/1134 needs to be solved before this one can be merged.

**Comment by nanory** [comment]
There will be a meeting in which we will discuss all problems we have identified so far and needs to be done to resolve this issue (See [here](https://github.com/gardenlinux/gardenlinux/issues/1191#issuecomment-1273044750))

**Review by nanory — APPROVED** [review]

**Comment by nanory** [comment]
The change looks good to me. I tested it with `aws-dev` by using the `_readonly` feature on top and all the services came up. Looks pretty promising from my point of view.

**Review by nanory — CHANGES_REQUESTED** [review]
Before merging, let @MrBatschner do the gardener test.

**Review by MrBatschner — APPROVED** [review]
I just tested these changes with Gardener and everything looks very nice.
/unhold
/squash
/lgtm

File: features/gardener/file.include/usr/lib/tmpfiles.d/gardener.conf
Comment (MrBatschner) [inline features/gardener/file.include/usr/lib/tmpfiles.d/gardener.conf:5]: Please leave this file out, they are no longer part of Garden Linux and no longer required by Gardener.

File: features/gardener/file.include/usr/share/factory/var/lib/gardener-gardenlinux/01_configure_lsm.sh
Comment (MrBatschner) [inline features/gardener/file.include/usr/share/factory/var/lib/gardener-gardenlinux/01_configure_lsm.sh:1]: See above, no longer required by Gardener, so please leave this file out.

File: features/gardener/file.include/usr/share/factory/var/lib/gardener-gardenlinux/02_iptables_backend.sh
Comment (MrBatschner) [inline features/gardener/file.include/usr/share/factory/var/lib/gardener-gardenlinux/02_iptables_backend.sh:1]: See above, no longer required by Gardener, so please leave this file out.

File: features/gardener/file.include/usr/share/factory/var/lib/gardener-gardenlinux/restart_kernel.sh
Comment (MrBatschner) [inline features/gardener/file.include/usr/share/factory/var/lib/gardener-gardenlinux/restart_kernel.sh:1]: See above, no longer required by Gardener, so please leave this file out.

**Review by MrBatschner — CHANGES_REQUESTED** [review]

**Review by nanory — APPROVED** [review]

**Comment by nkraetzschmar** [comment]
replaced by #1435

**Pull request closed without merging**
